migration/src/m0000650_alter_advisory_tracking/update_deprecated_advisory.sql

```diff
@@ -0,0 +1,16 @@
+CREATE OR REPLACE FUNCTION update_deprecated_advisory(identifier_input TEXT DEFAULT NULL)
+    RETURNS VOID AS
+$$
+BEGIN
+    WITH MostRecent AS (SELECT DISTINCT ON (identifier) id
+                        FROM advisory
+                        WHERE identifier = COALESCE(identifier_input, identifier)
+                        ORDER BY identifier, modified DESC)
+    UPDATE advisory
+    SET deprecated = CASE
+                         WHEN id IN (SELECT id FROM MostRecent) THEN FALSE
+                         ELSE TRUE
+        END
+    WHERE identifier = COALESCE(identifier_input, identifier);
+END;
+$$ LANGUAGE plpgsql;
```

Comment:
there maybe a more efficient func eg ... is distinct really needed ? dont we want to just return 1 thing from this ? using **!=** operator instead of **IN** probably avoids temporary result set (though I would have to check with explain analyze ... kinda hard with functions). 

```
CREATE OR REPLACE FUNCTION update_deprecated_advisory(identifier_input TEXT DEFAULT NULL)
    RETURNS VOID AS
$$
BEGIN
    UPDATE advisory
    SET deprecated = (id != (SELECT id
                             FROM advisory a
                             WHERE a.identifier = COALESCE(identifier_input, advisory.identifier)
                             ORDER BY a.modified DESC
                             LIMIT 1))
    WHERE identifier = COALESCE(identifier_input, identifier);
END;
$$ LANGUAGE plpgsql;
```